service/model.py

```diff
@@ -154,10 +155,26 @@ def find_or_404(cls, by_id):
 
     @classmethod
     def find_by_id_status(cls, by_id: int, by_status=True):
-        """ Find active recommendations of a product [id] """
+        """ Find [status: active/inactive] recommendations of a [product: id] """
+        if not by_id or not isinstance(by_id, int):
+            raise TypeError("by_id is not of type int")
+        if not isinstance(by_status, bool):
+            raise TypeError("by_status is not of type bool")
+
         cls.logger.info("Processing lookup for id %s with status %s", by_id, by_status)
         return cls.query.filter(cls.id==by_id, cls.status==by_status)
 
+    @classmethod
+    def find_by_id_type(cls, by_id: int, by_type: int):
+        """ Find recommendations of a [product: id] with [type: typeid] """
+        if not by_id or not isinstance(by_id, int):
+            raise TypeError("by_id is not of type int")
+        if not 1 <= by_type <= 3:
```

Comment:
Awesome, great work. One last thing, could you add integer type check for "by_type" also, similar to what you did for by_id?

Reply:
Yes sure I just added the type check. Sorry for the delay.